libtransact/src/context/manager/thread/mod.rs

```diff
@@ -0,0 +1,432 @@
+                    handler_receiver.recv()?
+                {
+                    Ok(())
+                } else {
+                    Err(ContextManagerError::InternalError(Box::new(
+                        ContextManagerCoreError::CoreReceiveError(RecvError),
+                    )))
+                }
+            }
+            Err(err) => Err(ContextManagerError::InternalError(Box::new(
+                ContextManagerCoreError::HandlerSendError(err),
+            ))),
+        }
+    }
+
+    /// Adds data, represented as `Vec<u8>`, to the specified context's list of data.
+    ///
+    pub fn add_data(
+        &self,
+        context_id: &ContextId,
+        data: Vec<u8>,
+    ) -> Result<(), ContextManagerError> {
+        let (handler_sender, handler_receiver) = mpsc::channel();
+
+        match self.core_sender.send(ContextOperationMessage::AddData {
+            context_id: *context_id,
+            data,
+            handler_sender,
+        }) {
+            Ok(_) => {
+                if let ContextOperationResponse::ValidResult { result: None } =
+                    handler_receiver.recv()?
+                {
+                    Ok(())
+                } else {
+                    Err(ContextManagerError::InternalError(Box::new(
+                        ContextManagerCoreError::CoreReceiveError(RecvError),
+                    )))
+                }
+            }
+            Err(err) => Err(ContextManagerError::InternalError(Box::new(
+                ContextManagerCoreError::HandlerSendError(err),
+            ))),
+        }
+    }
+}
+
+impl ContextLifecycle for ContextManager {
+    /// Creates a new context, returning the resulting ContextId
+    ///
+    fn create_context(
+        &mut self,
+        dependent_contexts: &[ContextId],
+        state_id: &str,
+    ) -> Result<ContextId, ContextManagerError> {
+        let (handler_sender, handler_receiver) = mpsc::channel();
+
+        match self
+            .core_sender
+            .send(ContextOperationMessage::CreateContext {
+                dependent_contexts: dependent_contexts.to_vec(),
+                state_id: state_id.to_string(),
+                handler_sender,
+            }) {
+            Ok(_) => {
+                if let ContextOperationResponse::ValidResult {
+                    result: Some(ContextOperationResult::CreateContext { context_id }),
+                } = handler_receiver.recv()?
+                {
+                    Ok(context_id)
+                } else {
+                    Err(ContextManagerError::InternalError(Box::new(
+                        ContextManagerCoreError::CoreReceiveError(RecvError),
+                    )))
+                }
+            }
+            Err(err) => Err(ContextManagerError::InternalError(Box::new(
+                ContextManagerCoreError::HandlerSendError(err),
+            ))),
+        }
+    }
+
+    fn drop_context(&mut self, _context_id: ContextId) {
+        unimplemented!();
```

Comment:
Could this be implemented in this PR?

Reply:
It certainly could be, I think before there was still a question on how exactly it would drop the contexts that I don't think has been resolved (At least from what I've heard)